test/test_script.py

```diff
@@ -1171,10 +1171,11 @@ def test_cmd_datetime(self):
             areg = r"^Wrong number of arguments for \$datetime: Expected between 0 and 1, "
             with self.assertRaisesRegex(ScriptError, areg):
                 self.parser.eval("$datetime(abc,def)")
-            # Tests with invalid format code (ends with % failes on all platforms)
-            areg = r"^\$datetime:\d+:\d+: Unsupported format code"
-            with self.assertRaisesRegex(ScriptRuntimeError, areg):
-                self.parser.eval(r"$datetime(\%)")
+            # The following is disabled because it only raises the exception for Python < 3.7 on Ubuntu
+            # # Tests with invalid format code (ends with % failes on all platforms)
+            # areg = r"^\$datetime:\d+:\d+: Unsupported format code"
+            # with self.assertRaisesRegex(ScriptRuntimeError, areg):
+            #     self.parser.eval(r"$datetime(\%)")
```

Comment:
For reference:
https://bugs.python.org/issue35066
https://github.com/python/cpython/pull/11550

Reply:
You could split this to another test and check if ending '%' causes ValueError or not to skip it or not.

Something like (untested):

```python

    def test_cmd_datetime_ending_percent(self):
        try:
            datetime.datetime.now().strftime('%')
        except ValueError:
            pass
        else:
            self.skipTest('datetime module supports trailing %')
        # Save original datetime object and substitute one returning
        # a fixed now() value for testing.
        original_datetime = datetime.datetime
        datetime.datetime = _DateTime

        try:
            context = Metadata()
            context["foo"] = "%Y%m%d%H%M%S.%f"
            # Tests with invalid format code (ends with % failes on all platforms)
            areg = r"^\$datetime:\d+:\d+: Unsupported format code"
            with self.assertRaisesRegex(ScriptRuntimeError, areg):
                self.parser.eval(r"$datetime(\%)")
        except (AssertionError, ValueError, IndexError) as err:
            raise err
        finally:
            # Restore original datetime object
            datetime.datetime = original_datetime
```
```

Reply:
I was thinking about checking in `func_datetime` and then forcing it to raise a `ValueError` but that just wouldn't be right.  I really like your solution, and will see about incorporating it.